src/components/dateTimePicker/index.js

```diff
@@ -0,0 +1,207 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet} from 'react-native';
+import RNDateTimePicker from '@react-native-community/datetimepicker';
+import {Constants} from '../../helpers';
+import {Colors} from '../../style';
+import Assets from '../../assets';
+import {BaseComponent} from '../../commons';
+import {TextField} from '../inputs';
+import Dialog from '../dialog';
+import View from '../view';
+import Button from '../button';
+
+
+const MODES = {
+  DATE: 'date',
+  TIME: 'time'
+};
+
+/*eslint-disable*/
+/**
+ * @description: Date and Time Picker Component that wraps RNDateTimePicker for date and time modes.
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/DateTimePickerScreen.js
+ * @important: DateTimePicker uses a native library. You MUST add and link the native library to both iOS and Android projects.
+ * @extends: @react-native-community/datetimepicker
+ * @extendslink: https://github.com/react-native-community/react-native-datetimepicker#react-native-datetimepicker
+ */
+/*eslint-enable*/
+
+class DateTimePicker extends BaseComponent {
+  static displayName = 'DateTimePicker';
+  static propTypes = {
+    ...TextField.propTypes,
+    /**
+     * The type of picker to display ('date' or 'time')
+     */
+    mode: PropTypes.oneOf(Object.values(MODES)),
+    /**
+     * The initial value to set the picker to. Defaults to device's date / time
+     */
+    value: PropTypes.instanceOf(Date),
+    /**
+     * The onChange callback
+     * NOTE: on Android called only after pressing 'Ok' button, on iOS called on every change in wheel picker
+     */
+    onChange: PropTypes.func,
+    /**
+     * The minimum date or time value to use
+     */
+    minimumDate: PropTypes.instanceOf(Date),
+    /**
+     * The maximum date or time value to use
+     */
+    maximumDate: PropTypes.instanceOf(Date)
+  }
+
+  static defaultProps = {
+    ...TextField.defaultProps,
+    mode: MODES.DATE
+  };
+
+  constructor(props) {
+    super(props);
+
+    const initialValue = props.value || new Date();
+    this.chosenDate = initialValue;
+
+    this.state = {
+      showExpandableModal: false,
+      chosenDate: initialValue
+    };
+  }
+
+  generateStyles() {
+    this.styles = createStyles(this.props);
+  }
+
+  setDate = (event, date) => {
+    if (date !== undefined) {
+      this.chosenDate = date;
+
+      if (Constants.isAndroid) {
+        this.setState({chosenDate: this.chosenDate, showExpandableModal: false});
+      }
+    }
+
+    _.invoke(this.props, 'onChange');
+  }
+
+  toggleExpandableModal = () => {
+    this.setState({showExpandableModal: !this.state.showExpandableModal});
+  };
+
+  onDonePressed = () => {
+    this.toggleExpandableModal();
+    setTimeout(() => {
```

Comment:
maybe instead of triggering  setState twice you can pass to `toggleExpandableModal` an extra state to update with

Reply:
Not sure I understand. Why do you say I'm triggering setState twice? toggleExpandableModal is not triggered for Android...

Reply:
because `this.toggleExpandableModal` calls setState
and right after that you call it again for setting the `chosenDate`

Reply:
Yes, but that's intentional. I need to close the modal and only then update the value of the TextField. That is also why I added the timeout around the setState() on the chosen date. Do you think there is a better way to time it?

Reply:
you can pass a callback to `this.toggleExpandableModal` that will be called after its `setState`. this way it will happen after for sure.

Reply:
Done.